jni/sequencercontroller.cpp

```diff
@@ -123,53 +124,75 @@ void SequencerController::setLoopRange( int aStartPosition, int aEndPosition, in
     updateStepsPerBar( aStepsPerBar );
 }
 
-void SequencerController::updateStepsPerBar( int aStepsPerBar )
+int SequencerController::getBufferPosition()
 {
-    if ( stepsPerBar != aStepsPerBar )
-    {
-        stepsPerBar = aStepsPerBar;
-        AudioEngine::beat_subdivision = aStepsPerBar / AudioEngine::time_sig_beat_unit;
-    }
+    return AudioEngine::bufferPosition;
 }
 
-void SequencerController::updateMeasures( int aValue, int aStepsPerBar )
+void SequencerController::setBufferPosition( int aPosition )
 {
-    AudioEngine::amount_of_bars      = aValue;
-    AudioEngine::max_step_position   = aStepsPerBar * AudioEngine::amount_of_bars;
-    AudioEngine::max_buffer_position = AudioEngine::bytes_per_bar * AudioEngine::amount_of_bars;
+    // keep position within the sequences range (see "setLoopRange")
 
-    updateStepsPerBar( aStepsPerBar );
+    if ( aPosition < AudioEngine::min_buffer_position )
+        aPosition = AudioEngine::min_buffer_position;
+
+    else if ( aPosition > AudioEngine::max_buffer_position )
+        aPosition = AudioEngine::max_buffer_position;
+
+    AudioEngine::bufferPosition = aPosition;
+    AudioEngine::stepPosition   = ( aPosition / AudioEngine::samples_per_bar ) * stepsPerBar;
+
+    Notifier::broadcast( Notifications::SEQUENCER_POSITION_UPDATED );
 }
 
-void SequencerController::rewind()
+int SequencerController::getSamplesPerBeat()
 {
-    setPosition( AudioEngine::min_buffer_position );
+    return AudioEngine::samples_per_beat;
 }
```

Comment:
New convenience methods regarding buffer position / sample sizes as these are related to the Sequencer, they should belong here.